sdk/aqueduct/artifacts/generic_artifact.py

```diff
@@ -62,7 +54,10 @@ def get(self, parameters: Optional[Dict[str, Any]] = None) -> Any:
         self._dag.must_get_artifact(self._artifact_id)
 
         if self._from_flow_run:
-            if self._execution_status != ExecutionStatus.SUCCEEDED:
+            if (
+                not self._execution_state
+                or self._execution_state.status != ExecutionStatus.SUCCEEDED
```

Comment:
Um shouldn't this return None if the artifact was deleted? The error message talks about "computed successfully!" which is true when status is Deleted.

Reply:
That's a good catch! It's interesting that we are not handling other state though. Following the same argument I think we should. I will just address the DELETE state in this PR though

Reply:
What do you mean by handling other state? None of the other states computed the artifact successfully.

Reply:
I mean for fetched workflow runs, it's reasonable to call `.get()` even if the artifact is with fail / cancel / pending state with less blocking behavior like throwing this error

Reply:
yeah I think we just wanted to disambiguate a failed artifact vs an operator that returned None.

Reply:
I see, in that way I will add a few lines to print warning message if the artifact is deleted